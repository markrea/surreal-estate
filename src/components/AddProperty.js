import React, { useState } from "react";
import "../styles/AddProperty.css";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      city: "Manchester",
      type: "Flat",
      bedrooms: "1",
      bathrooms: "1",
      price: "",
      email: "",
    },
  };

  const [fields, setFields] = useState(initialState.fields);

  const handleAddProperty = (event) => {
    event.preventDefault();
    console.log(fields);
  };

  const handleFieldChange = (event) => {
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  return (
    <div className="AddProperty">
      <form onSubmit={handleAddProperty}>
        <div className="input-fields">
        <label className="label" htmlFor="title">
          Property Description
          <input
            className="form-field"
            id="title"
            name="title"
            value={fields.title}
            onChange={handleFieldChange}
            placeholder="e.g. One bedroom Flat"
          />
        </label>

        <br />
        <label className="label" htmlFor="city">
          City
          <select
            className="form-field"
            id="city"
            name="city"
            value={fields.city}
            onChange={handleFieldChange}
          >
            <option value="Manchester">Manchester</option>
            <option value="Leeds">Leeds</option>
            <option value="Sheffield">Sheffield</option>
            <option value="Liverpool">Liverpool</option>
          </select>
        </label>

        <br />
        <label className="label" htmlFor="Type">
          Type
          <select
            className="form-field"
            id="type"
            name="type"
            value={fields.type}
            onChange={handleFieldChange}
          >
            <option value="Flat">Flat</option>
            <option value="Detached">Detached</option>
            <option value="Semi-Detached">Semi-Detached</option>
            <option value="Terraced">Terraced</option>
            <option value="End of Terrace">End of Terrace</option>
            <option value="Cottage">Cottage</option>
            <option value="Bungalow">Bungalow</option>
          </select>
        </label>
        <br />
        <label className="label" htmlFor="bedrooms">
          Bedrooms
          <select
            className="form-field"
            id="bedrroms"
            name="bedrooms"
            value={fields.bedrooms}
            onChange={handleFieldChange}
          >
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
            <option value="4">4</option>
            <option value="5">5</option>
            <option value="6+">6+</option>
          </select>
        </label>
        <br />
        <label className="label" htmlFor="bathrooms">
          Bathrooms
          <select
            className="form-field"
            id="bathrooms"
            name="bathrooms"
            value={fields.bathrooms}
            onChange={handleFieldChange}
          >
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
            <option value="4+">4+</option>
          </select>
        </label>
        <br />
        <label className="label" htmlFor="price">
          Price
          <input
            className="form-field"
            id="price"
            name="price"
            value={fields.price}
            onChange={handleFieldChange}
            type="number"
            placeholder="Â£"
          />
        </label>
        <br />
        <label  className="label" htmlFor="email">
          Email
          <input
            className="form-field"
            id="email"
            name="email"
            value={fields.email}
            onChange={handleFieldChange}
            type="email"
            placeholder="e.g. yourname@account.com"
          />
        </label>
        
        </div>
        <div>
          <button type="submit">Add</button>
        </div>
      </form>
    </div>
  );
};

export default AddProperty;
