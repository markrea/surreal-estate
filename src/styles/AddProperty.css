.AddProperty {
    padding: 5px;

    display: flex;
    flex-direction: column;
    height: 100vh;
    align-items: center;
    font-size: 3vw;
    margin: 0 auto;
}



.input-fields { 
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
  
   

}

/* form * {
    box-sizing: border-box;
} */
 .label {
    display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 5px;
  max-width: 300px;;
 }

 input, select {
   
    align-items: center;
    margin: 5px;
    width: 100%;
    /* font-size: 3vw; */
  }

  /* .form-field {
    float: left;
    width: 75%;
    margin-top: 6px;
  } */
  button { 
      margin: 0 auto; 
      display: block;
      width: 30vw;
      font-size: 4vw;
  }

  @media screen and (min-width: 590px) {
    button {
        font-size: 25px;
    }

    .AddProperty {
        font-size: 25px;
    }
  }