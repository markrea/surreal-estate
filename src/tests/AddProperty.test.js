import React from "react";
import { render } from "@testing-library/react";
import AddProperty from "../components/AddProperty";

describe("AddProperty", () => {
  test("component renders correctly", () => {
    const { asFragment } = render(<AddProperty />);
    const component = asFragment();

    expect(component).toMatchSnapshot();
  });

  test("renders AddProperty", () => {
      const { getByLabelText, getByRole } = render(<AddProperty />);
      
      expect(getByLabelText("Property Description")).toBeInTheDocument();
      expect(getByLabelText("City")).toBeInTheDocument();
      expect(getByLabelText("Type")).toBeInTheDocument();
      expect(getByLabelText("Bedrooms")).toBeInTheDocument();
      expect(getByLabelText("Bathrooms")).toBeInTheDocument();
      expect(getByLabelText("Price")).toBeInTheDocument();
      expect(getByLabelText("Email")).toBeInTheDocument();
      expect(getByRole("button")).toBeInTheDocument();
  });
 
});
