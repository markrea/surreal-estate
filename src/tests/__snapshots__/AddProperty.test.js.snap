// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="AddProperty"
  >
    <form>
      <div
        class="input-fields"
      >
        <label
          class="label"
          for="title"
        >
          Property Description
          <input
            class="form-field"
            id="title"
            name="title"
            placeholder="e.g. One bedroom Flat"
            value=""
          />
        </label>
        <br />
        <label
          class="label"
          for="city"
        >
          City
          <select
            class="form-field"
            id="city"
            name="city"
          >
            <option
              value="Manchester"
            >
              Manchester
            </option>
            <option
              value="Leeds"
            >
              Leeds
            </option>
            <option
              value="Sheffield"
            >
              Sheffield
            </option>
            <option
              value="Liverpool"
            >
              Liverpool
            </option>
          </select>
        </label>
        <br />
        <label
          class="label"
          for="Type"
        >
          Type
          <select
            class="form-field"
            id="type"
            name="type"
          >
            <option
              value="Flat"
            >
              Flat
            </option>
            <option
              value="Detached"
            >
              Detached
            </option>
            <option
              value="Semi-Detached"
            >
              Semi-Detached
            </option>
            <option
              value="Terraced"
            >
              Terraced
            </option>
            <option
              value="End of Terrace"
            >
              End of Terrace
            </option>
            <option
              value="Cottage"
            >
              Cottage
            </option>
            <option
              value="Bungalow"
            >
              Bungalow
            </option>
          </select>
        </label>
        <br />
        <label
          class="label"
          for="bedrooms"
        >
          Bedrooms
          <select
            class="form-field"
            id="bedrroms"
            name="bedrooms"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6+"
            >
              6+
            </option>
          </select>
        </label>
        <br />
        <label
          class="label"
          for="bathrooms"
        >
          Bathrooms
          <select
            class="form-field"
            id="bathrooms"
            name="bathrooms"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4+"
            >
              4+
            </option>
          </select>
        </label>
        <br />
        <label
          class="label"
          for="price"
        >
          Price
          <input
            class="form-field"
            id="price"
            name="price"
            placeholder="Â£"
            type="number"
            value=""
          />
        </label>
        <br />
        <label
          class="label"
          for="email"
        >
          Email
          <input
            class="form-field"
            id="email"
            name="email"
            placeholder="e.g. yourname@account.com"
            type="email"
            value=""
          />
        </label>
      </div>
      <div>
        <button
          type="submit"
        >
          Add
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
